lib/action_subscriber/publisher/async/in_memory_adapter.rb

```diff
@@ -92,11 +93,20 @@ def create_and_supervise_consumer!
             @consumer = create_consumer
             @supervisor = ::Thread.new do
               loop do
-                unless consumer.alive?
-                  # Why might need to requeue the last message.
-                  queue.push(@current_message) if @current_message.present?
-                  consumer.kill
-                  @consumer = create_consumer
+                begin
+                  unless consumer.alive?
+                    # Why might need to requeue the last message.
+                    queue.push(@current_message) if @current_message.present?
+                    consumer.kill
+                    @consumer = create_consumer
+                  end
+
+                  instrumentation_threadpool.async do
+                    payload = { :queue_size => size, :is_consumer_alive => consumer.alive? }
+                    ::ActiveSupport::Notifications.instrument "supervisor_tick.in_memory_publisher.action_subscriber", payload
+                  end
+                rescue
```

Comment:
@mmmries I just want to ensure that the supervisor never dies. So this ensures things keep ticking. I agree it's a bit weird, though.